ee/ucp/admin/configure/route-reflector-deployment.md

```diff
@@ -0,0 +1,210 @@
+
+    a) You can set the environment variables
+    
+       ```
+       export ETCD_ENDPOINTS=127.0.0.1:12379 
+       export ETCD_KEY_FILE=/var/lib/docker/volumes/ucp-node-certs/_data/key.pem 
+       export ETCD_CA_CERT_FILE=/var/lib/docker/volumes/ucp-node-certs/_data/ca.pem 
+       export ETCD_CERT_FILE=/var/lib/docker/volumes/ucp-node-certs/_data/cert.pem 
+       ```
+       
+    b) You can create a configuration file.
+       ```calicoctl``` by default looks at ```/etc/calico/calicoctl.cfg``` for the configuration. For custom 
+       configuration file location, use ```--config``` with ```calicoctl```
+       
+       ```calicoctl.cfg:```
+       ```
+       apiVersion: projectcalico.org/v3
+       kind: CalicoAPIConfig
+       metadata:
+       spec:
+         datastoreType: "etcdv3"
+         etcdEndpoints: "127.0.0.1:12379"
+         etcdKeyFile: "/var/lib/docker/volumes/ucp-node-certs/_data/key.pem"
+         etcdCertFile: "/var/lib/docker/volumes/ucp-node-certs/_data/cert.pem"
+         etcdCACertFile: "/var/lib/docker/volumes/ucp-node-certs/_data/ca.pem"
+       ```
+   
+3) Identify the nodes where the route reflector needs to be deployed. Please note, this node will not be able to schedule
+   any kubernetes workload. However you will continue to have kubernetes system containers if present on this node.
+
+   In this example node ```ubuntu-0``` is being tainted. Use the client bundle and run:
+   
+   ```kubectl taint node ubuntu-0 com.docker.ucp.kubernetes.calico/route-reflector=true:NoSchedule```
+   
+   This will need to be done on every node where the calico route reflector pod will be deployed.
+
+4) Add labels to the same node. The labels will be used to match by the scheduler to deploy the calico route reflector pods.
+   Use the client bundle and run:
+
+   ```kubectl label nodes ubuntu-0 com.docker.ucp.kubernetes.calico/route-reflector=true ```
+
+   This will need to be done on every node where the calico route reflector pod will be deployed.
+
+5) Deploy the calico route reflector Daemonset. The calico-rr pods will be deployed on all nodes that have the taint
+   and the node labels.
+   
+   ```calico-rr.yaml```
+   
+   ```
+   kind: DaemonSet
+   apiVersion: extensions/v1beta1
+   metadata:
+     name: calico-rr
+     namespace: kube-system
+     labels:
+       app: calico-rr
+   spec:
+     updateStrategy:
+       type: RollingUpdate
+     selector:
+       matchLabels:
+         k8s-app: calico-rr
+     template:
+       metadata:
+         labels:
+           k8s-app: calico-rr
+         annotations:
+           scheduler.alpha.kubernetes.io/critical-pod: ''
+       spec:
+         tolerations:
+           - key: com.docker.ucp.kubernetes.calico/route-reflector
+             value: "true"
+             effect: NoSchedule
+         hostNetwork: true
+         containers:
+           - name: calico-rr
+             image: calico/routereflector:v0.6.1
```

Comment:
Since UCP uses Calico 3.0.1, are there any compatibility requirements that users should be aware?

Reply:
not at this point. We can document when it becomes a compatibility requirement